recipes/gn/all/conanfile.py

```diff
@@ -1,128 +1,165 @@
+        env.vars(self).save_script("conanbuild_gn")
+
+        if is_msvc(self):
+            vcvars = VCVars(self)
+            vcvars.generate()
+
+        configure_args = [
+            "--no-last-commit-position",
+            f"--host={self._gn_platform}",
+        ]
+        if self.settings.build_type in ["Debug", "RelWithDebInfo"]:
+            configure_args.append("-d")
+        save(self, os.path.join(self.source_folder, "configure_args"), " ".join(configure_args))
 
     def build(self):
-        with tools.chdir(self._source_subfolder):
-            with self._build_context():
-                # Generate dummy header to be able to run `build/ben.py` with `--no-last-commit-position`. This allows running the script without the tree having to be a git checkout.
-                tools.save(os.path.join("src", "gn", "last_commit_position.h"),
-                           textwrap.dedent("""\
-                                #pragma once
-                                #define LAST_COMMIT_POSITION "1"
-                                #define LAST_COMMIT_POSITION_NUM 1
-                                """))
-                conf_args = [
-                    "--no-last-commit-position",
-                    "--host={}".format(self._to_gn_platform(self.settings.os, self.settings.compiler)),
-                ]
-                if self.settings.build_type == "Debug":
-                    conf_args.append("-d")
-                self.run("{} build/gen.py {}".format(sys.executable, " ".join(conf_args)), run_environment=True)
-                # Try sleeping one second to avoid time skew of the generated ninja.build file (and having to re-run build/gen.py)
-                time.sleep(1)
-                build_args = [
-                    "-C", "out",
-                    "-j{}".format(tools.cpu_count()),
-                ]
-                self.run("ninja {}".format(" ".join(build_args)), run_environment=True)
+        with chdir(self, self.source_folder):
+            # Generate dummy header to be able to run `build/gen.py` with `--no-last-commit-position`.
+            # This allows running the script without the tree having to be a git checkout.
+            save(self, os.path.join(self.source_folder, "src", "gn", "last_commit_position.h"),
+                textwrap.dedent("""\
+                    #pragma once
+                    #define LAST_COMMIT_POSITION "1"
+                    #define LAST_COMMIT_POSITION_NUM 1
+                    """),
+            )
+
+            # Disable GenerateLastCommitPosition()
+            replace_in_file(self, os.path.join(self.source_folder, "build/gen.py"),
+                            "def GenerateLastCommitPosition(host, header):",
+                            "def GenerateLastCommitPosition(host, header):\n  return")
+
+            # Try sleeping to avoid time skew of the generated ninja.build file (and having to re-run build/gen.py)
+            err = None
+            for sleep in [0, 1, 2, 4, 8]:
```

Comment:
I need some context on this one. If the original sleep already raised some eyebrows, now with the loop it really needs some explanation 🤯

Reply:
@uilianries Unfortunately, it's not clear anymore what error to expect in this specific case, plus capturing the output without silencing it is kind of a pain, so I simply dropped this workaround for now.